### ISSUE TITLE
the path Hog_feature.Hog_feature.hog

### ISSUE THREAD

**Issue by zyh778** [MubarizZaffar/CoHOG_Results_RAL2019#2]
"Hello, I have recently read your paper 'CoHOG: A Light-Weight, Compute-Efficient, and Training-Free Visual Place Recognition Technique for Changing Environments' and am very interested in it. I would like to run the open-source code you provided on GitHub. However, in the main.py file, I found these two lines of code for importing packages:

from Hog_feature.Hog_feature.hog import initialize
from Hog_feature.Hog_feature.hog import extract

I am unable to locate the files at the path Hog_feature.Hog_feature.hog. Could you please advise how to resolve this issue?"

**Comment by MubarizZaffar** [comment]
Hey, thanks for your interest. Yes, it's already in the repo. See [here](https://github.com/MubarizZaffar/CoHOG_Results_RAL2019/blob/master/CoHOG_Python_Implementation/Hog_feature.zip).

**Comment by zyh778** [comment]
I found these files in this zip, where hog is a .so file, do I need to import this so file first before I  import the package?
![image](https://github.com/user-attachments/assets/f96efa44-c4b6-4b6c-b932-567ddacdfe63)
![image](https://github.com/user-attachments/assets/7f6ca9cc-d61a-4837-9e50-bc05c4b76faa)

**Comment by MubarizZaffar** [comment]
No, just extract the Zip folder in the directory. The paths will work as is, you don't need to change anything, just need to extract the Zip folder.

A ._so_ is a Python library file, you can import it like a typical Python module.

**Comment by zyh778** [comment]
![image](https://github.com/user-attachments/assets/87f00187-98e7-4a19-ac25-18000130f2a1)
![image](https://github.com/user-attachments/assets/852fbdcc-5227-43ba-8320-08c8fd5af51a)
But I get the above problem when importing, it also shows that my python version is inconsistent, which version of python is compatible please!

**Comment by MubarizZaffar** [comment]
Looks like a Cython compilation error. Perhaps you are running the Python 2 version with Python 3? There is also a separate Python3 compiled version available.

Even if you don't want to use the Cython compiled libraries, the source code is also available for the library in this repo.

**Comment by zyh778** [comment]
Thank you very much for your reply again.
At first I did use python3.10 to run python2, but I immediately found the problem and used python3.10 to run python3, but the problem still occurs.
